apis/kueue/v1beta1/provisioningrequestconfig_types.go

```diff
@@ -21,10 +21,15 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type ProvisioningRequestConfigMergePodSetPolicy string
+
 const (
 	// ProvisioningRequestControllerName is the name used by the Provisioning
 	// Request admission check controller.
 	ProvisioningRequestControllerName = "kueue.x-k8s.io/provisioning-request"
+
+	ProvisioningDirectives ProvisioningRequestConfigMergePodSetPolicy = "ProvisioningDirectives"
```

Comment:
The merge is done based on the workload definition. In the definition there is usually nothing about provisioning. There are scheduling requirements (node selectors, affinity, tolerations etc) which may result in provisioning some nodes. However provisioning itself is usually not the intent of the user. They just provide constraints. Thus I have mixed feelings about this name. What about "IdenticalSchedulingRequirements" ?

Reply:
IIUC, we use the containers as identification when merging multiple PodSets. It indicates env, command, and others will be used. And those are not related to scheduling. So, I intentionally avoid the "scheduling" word.

>  However provisioning itself is usually not the intent of the user. 

Could you elaborate this one? I thought we can say `IdenticalProvisioningDirectives` to represent "The workloads with the same Provisioning Directives are considered to be able to request additional Nodes at once"

Reply:
Maybe @mwielgus can elaborate more, but I synced with Marcin, and there are two main points:
1. ProvisioningDirectives is not a known term current, also SchedulingDirectives are not widely known to users, and this is user / admin-facing API
2. creating PodTemplates users don't think about "Provisioning", it is job of the admin to define the ResourceFlavors. Depending on the selected resource flavor the Pod might be provisioning or not (may land on Reservations or Provisioning RF). So, the fields that user specifies in the template are the user's "scheduling requirements", not "provisioning requirements".

I think these are valid points. Having said that I'm personally ok with either `IdenticalProvisioningDirectives` or `IdenticalSchedulingRequirements`.

Reply:
I'm OK with IndenticalWorkloadSchedulingRequirements

Reply:
> I'm OK with IndenticalWorkloadSchedulingRequirements

LGTM, thanks!